### PR TITLE
0.3.0 release

### PR DESCRIPTION
**Pull request by naterush**
see release notes :D

### CODE REVIEW

File: casper/protocols/sharding/forkchoice.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    """Returns a dict of block => weight"""
+    scores = dict()
+
+    for validator, current_block in latest_messages.items():
+        if not current_block.on_shard(shard_id):
+            continue
+
+        while current_block and current_block != starting_block:
+            scores[current_block] = scores.get(current_block, 0) + validator.weight
+            current_block = current_block.prev_block(shard_id)
```
Comment (drewstone) [inline casper/protocols/sharding/forkchoice.py]: Can previous blocks occur in multiple shards? Or is this a unique selection, i.e. each block's previous history occurs in exactly 1 shard

File: casper/protocols/sharding/sharding_plot_tool.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            new_messages = []
+
+        self._update_new_justifications(new_messages)
+        self._update_blockchain(new_messages)
+        self._update_block_fault_tolerance()
+        self._update_message_labels(new_messages)
+
+    def plot(self):
+        """Builds relevant edges to display and creates next viewgraph using them"""
+        return
```
Comment (drewstone) [inline casper/protocols/sharding/sharding_plot_tool.py]: Typo perhaps? Or do you want to return immediately

Comment (djrtwo) [reply casper/protocols/sharding/sharding_plot_tool.py]: Currently don't have graph plotting on the sharding consensus, just some terminal printing. So this is a stub

File: casper/protocols/sharding/forkchoice.py (hunk 1 above)
Comment (djrtwo) [reply casper/protocols/sharding/forkchoice.py]: A merge-block has prev_block's of multiple shards (2 in this current implementation). This is the core mechanism that provides atomicity of shard transactions and builds up the fork choice rule in binary sharding.

Comment (drewstone) [reply casper/protocols/sharding/forkchoice.py]: @djrtwo that makes sense, thanks!

**Pull request merged**